common/src/backends/otlp_tracing_bridge.rs

```diff
@@ -0,0 +1,154 @@
+// Taken from
+// https://github.com/open-telemetry/opentelemetry-rust/blob/e640051b8bd5d56bb058ec6caabadf2bee5244a9/opentelemetry-appender-tracing/src/layer.rs,
+// waiting for https://github.com/open-telemetry/opentelemetry-rust/pull/1394 to be merged.
+// This is under Apache License 2.0
+
+use opentelemetry::{
+    logs::{LogRecord, Logger, LoggerProvider, Severity, TraceContext},
+    trace::{SpanContext, TraceFlags, TraceState},
+};
+use std::borrow::Cow;
+use tracing_core::{Level, Subscriber};
+use tracing_opentelemetry::OtelData;
+use tracing_subscriber::{registry::LookupSpan, Layer};
+
+const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
+
+/// Visitor to record the fields from the event record.
+struct EventVisitor<'a> {
+    log_record: &'a mut LogRecord,
+}
+
+impl<'a> tracing::field::Visit for EventVisitor<'a> {
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        if field.name() == "message" {
+            self.log_record.body = Some(format!("{value:?}").into());
+        } else if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), format!("{value:?}").into()));
+        } else {
+            let vec = vec![(field.name().into(), format!("{value:?}").into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_str(&mut self, field: &tracing_core::Field, value: &str) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.to_owned().into()));
+        } else {
+            let vec = vec![(field.name().into(), value.to_owned().into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_bool(&mut self, field: &tracing_core::Field, value: bool) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_f64(&mut self, field: &tracing::field::Field, value: f64) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    fn record_i64(&mut self, field: &tracing::field::Field, value: i64) {
+        if let Some(ref mut vec) = self.log_record.attributes {
+            vec.push((field.name().into(), value.into()));
+        } else {
+            let vec = vec![(field.name().into(), value.into())];
+            self.log_record.attributes = Some(vec);
+        }
+    }
+
+    // TODO: Remaining field types from AnyValue : Bytes, ListAny, Boolean
```

Comment:
Do we want to cover this TODO before upgrading from draft PR? I guess there will be compile time errors in case the types need to be implemented, so not sure if it is worth it in case we don't have errors at this point.